unlock-app/src/components/content/DashboardContent.js

```diff
@@ -35,35 +55,53 @@ DashboardContent.propTypes = {
   account: UnlockPropTypes.account,
   network: UnlockPropTypes.network.isRequired,
   lockFeed: PropTypes.arrayOf(UnlockPropTypes.lock),
+  formIsVisible: PropTypes.bool.isRequired,
+  showForm: PropTypes.func.isRequired,
+  hideForm: PropTypes.func.isRequired,
 }
 
 DashboardContent.defaultProps = {
   lockFeed: [],
   account: null,
 }
 
-export const mapStateToProps = state => {
+export const mapStateToProps = ({
+  transactions,
+  locks,
+  account,
+  network,
+  lockFormStatus: { visible },
+}) => {
   // We want to display newer locks first, so sort the locks by blockNumber in descending order
   const locksComparator = (a, b) => {
     // Newly created locks may not have a transaction associated just yet
     // -- those always go right to the top
-    if (!state.transactions[a.transaction]) {
+    if (!transactions[a.transaction]) {
       return -1
     }
-    if (!state.transactions[b.transaction]) {
+    if (!transactions[b.transaction]) {
       return 1
     }
     return (
-      state.transactions[b.transaction].blockNumber -
-      state.transactions[a.transaction].blockNumber
+      transactions[b.transaction].blockNumber -
+      transactions[a.transaction].blockNumber
     )
   }
-  const lockFeed = Object.values(state.locks).sort(locksComparator)
+  const lockFeed = Object.values(locks).sort(locksComparator)
   return {
-    account: state.account,
-    network: state.network,
+    account: account,
+    network: network,
     lockFeed,
+    formIsVisible: visible,
   }
 }
 
-export default connect(mapStateToProps)(DashboardContent)
+const mapDispatchToProps = dispatch => ({
```

Comment:
is it possible in ts to do this instead?
`const mapDispatchToProps = { showForm, hideForm }`

redux will auto-create the action dispatchers for us

Reply:
In a previous PR I was told to do it manually for consistency with the other instances of mapDispatchToProps